gateway/service/src/main/java/org/eclipse/ditto/gateway/service/endpoints/routes/thingsearch/ThingSearchRoute.java

```diff
@@ -79,10 +79,28 @@ public Route buildSearchRoute(final RequestContext ctx, final DittoHeaders ditto
      */
     private Route countThings(final RequestContext ctx, final DittoHeaders dittoHeaders) {
         // GET things/count?filter=<filterString>&namespaces=<namespacesString>
-        return get(() -> thingSearchParameterOptional(params -> handlePerRequest(ctx,
-                CountThings.of(calculateFilter(params.get(ThingSearchParameter.FILTER)),
-                        calculateNamespaces(params.get(ThingSearchParameter.NAMESPACES)),
-                        dittoHeaders))));
+        // POST things/count?namespaces=<namespacesString>
+        return concat(
+                get(() -> thingSearchParameterOptional(
+                        params -> handlePerRequest(ctx,
+                            CountThings.of(calculateFilter(params.get(ThingSearchParameter.FILTER)),
+                            calculateNamespaces(params.get(ThingSearchParameter.NAMESPACES)),
+                                    dittoHeaders)))),
+                post(() -> thingSearchParameterOptional(
+                        params -> ensureMediaTypeJsonWithFallbacksThenExtractDataBytes(
```

Comment:
By using `ensureMediaTypeJsonWithFallbacksThenExtractDataBytes` you enforce the POST payload to be of `application/json` - do you really want that?
Did you test that?

To me, this looks like it could not be what is wanted - as the `filter` below inserted into `CountThings` is certainly not a Json, but a plain text string.

I would expect the content-type to be `application/x-www-form-urlencoded` instead  and that e.g. also query params would be provided all as form values - instead of mixing those.